### PR TITLE
Draft: Sanity check build with JDK 18 on CI

### PR DESCRIPTION
**Pull request by szpak**
Add basic JDK 18 tests on CI with Temurin.

Unfortunately, Gradle 7.4.1/2 [refuses](https://github.com/spockframework/spock/runs/5960598888?check_suite_focus=true#step:4:43) to play with JDK 18:
```
 * What went wrong:
A problem occurred evaluating project ':spock-specs'.
> Could not resolve all dependencies for configuration ':spock-specs:jacocoAgentRuntime'.
   > Failed to query the value of task ':spock-specs:compileGroovy' property 'javaLauncher'.
      > No compatible toolchains found for request filter: {languageVersion=18, vendor=any, implementation=vendor-specific} (auto-detect true, auto-download false)
```

However, it builds fine locally with JDK 18. Is it "just" a toolchain detection issue which could be easily overridden or Gradle 7.5 is a must?

### CODE REVIEW

File: .github/actions/setup-jdks/action.yml [hunk 1]
```diff
@@ -9,10 +9,16 @@ runs:
...
-      if: inputs.additional-java-version != 8
+      if: inputs.additional-java-version != 8 && inputs.additional-java-version < 18
       with:
         distribution: 'adopt'
         java-version: ${{ inputs.additional-java-version }}
+    - name: 'Set up JDK ${{ inputs.additional-java-version }}'
+      uses: actions/setup-java@v2
+      if: inputs.additional-java-version > 17
+      with:
+        distribution: 'temurin'
```
Comment (leonard84) [inline .github/actions/setup-jdks/action.yml:20]: I would switch everything to temurin if possible, as adopt is deprecated.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,7 +1,7 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-# https://gradle.org/release-checksums/
-distributionSha256Sum=e5444a57cda4a95f90b0c9446a9e1b47d3d7f69057765bfb54bd4f482542d548
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.1-bin.zip
+#distributionSha256Sum=29e49b10984e585d8118b7d0bc452f944e386458df27371b49b4ac1dec4b7fda
+#distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
+distributionUrl=https://services.gradle.org/distributions-snapshots/gradle-7.5-20220427150934+0000-bin.zip
```
Comment (szpak) [inline gradle/wrapper/gradle-wrapper.properties:5]: I gave 7.5-SNAPSHOT a try, but it still fails with a toolchain and JDK 18:
https://github.com/spockframework/spock/runs/6201180386?check_suite_focus=true#step:4:49

@leonard84 Do you recognize any related open issue in Gradle to make it fixed before 7.5 is released?

Comment (szpak) [reply gradle/wrapper/gradle-wrapper.properties]: It was not a problem with Gradle, just with settings `org.gradle.java.installations.fromEnv` to handle the  JDK18 env variable... I added a comment in CI configuration to remember about that.

**Comment by szpak** [comment]
While the toolchain is fixed, one test fails with JDK18 (Temurin 18.0.1+10) on CI:
https://github.com/spockframework/spock/runs/6235842140?check_suite_focus=true#step:4:152

but locally everything passes with JDK 18 (OpenJDK 18+37 by RedHat)...

**Comment by Vampire** [comment]
As we have test run for Java 21 now, this PR is obsolete, isn't it?

**Pull request closed without merging**